contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,128 @@
+1. [Understanding Performance Collection Methods](https://msdn.microsoft.com/en-us/library/dd264994.aspx)
+
+# Performance Profiling For The First Time
+
+Most profilers are similar in their functionalities, user interface, and use of 
+technical terms. This tutorial demonstrates using the profiler included in 
+[Visual Studio 2015](https://www.visualstudio.com/downloads/).
+
+> You are encouraged to adapt this tutorial to your preferred profiling 
+tools on your own project. <br>
+
+## 1. Run a performance profiling session
+
+To start a performance profiling session, follow this guide on 
+[Creating and running a performance session](https://msdn.microsoft.com/en-us/library/ms182372.aspx#Anchor_0).
+
+## 2. View the performance report
+
+The profiler will generate the performance report after the profiling session, which you 
+can explore on your own. *Inclusive and exclusive data* provides meaningful information 
+about the execution time of each function.
+
+* **Inclusive samples (or elapsed time)**: Collected during execution of the function itself and all functions it calls
+* **Exclusive samples (or elapsed time)**: Collected during execution of the function itself only
+
+> The Visual Studio Profiler Team Blog has a [good explanation on inclusive and exclusive data values](https://blogs.msdn.microsoft.com/profiler/2004/06/09/what-are-exclusive-and-inclusive/).
+
+## 3. Trace down the problem
+
+You can identify the performance issues using the performance report. The two 
+useful information analyzed by the profiler are shown in the `Summary` view.
+
+* **Functions With Most Individual Work**: The functions which took up the most execution time *(exclusive samples)*
+* **Hot Path**: The branch of the call tree which took up the most execution time *(inclusive samples)*
+
+To locate performance issues quickly, the **Functions With Most Individual Work** provides 
+a list of functions which are usually candidates for optimization.
+
+If you would like to trace down to the problem more carefully, the **Hot Path** is a good 
+starting point. This will familiarize you about the most expensive execution path taken 
+by your program. 
+
+> You can follow this [walkthrough](https://msdn.microsoft.com/en-us/library/ms182398.aspx)
+to learn how to identify performance problems.
+
+## 4. Identify areas for improvements upwards
+
+You have identified the problem, and may now want to optimize the code in the 
+function body. But before that, here's a final tip: 
+
+> It is sometimes possible (and easier) to optimize by reducing the number of calls to 
+that function in its calling functions.
+
+### Resources:
+1. [Beginners Guide to Performance Profiling](https://msdn.microsoft.com/en-us/library/ms182372.aspx)
+1. [What are Exclusive and Inclusive?!](https://blogs.msdn.microsoft.com/profiler/2004/06/09/what-are-exclusive-and-inclusive/)
+1. [Walkthrough: Identifying Performance Problems](https://msdn.microsoft.com/en-us/library/ms182398.aspx)
```

Comment:
Perhaps you can distinguish between these two types of resources:
1. resources you used when writing this article: explain in what way you used each resource
2. recommended resources for the reader to learn more: explain what additional knowledge the reader can get from those resources